Mhv 55030 update list cards

robertroddev opened the pull request:
## Summary
Update list cards on prescriptions view. All updates are only shown if refill flag `mhv_medications_display_refill_content` is enabled.
## Related issue(s)
[Update list page cards](https://jira.devops.va.gov/browse/MHV-55030)

> Acceptance Criteria:
AC1: Add grey container with new content and primary green action link that takes users to the new refill page/flow
AC2: Remove all fill/refill buttons from the cards. Users will be required to use the new flow option to fill or refill their meds
AC3: Change style of sort button from primary to secondary supporting drawing users attention to the new CTA at top of page

## Testing done

Unit test, 
Ran coverage report
<img width="735" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/107279507/93d60b02-7be9-4f07-8294-3e2e22a9afeb">


## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |<img width="366" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/107279507/62d8c192-aa0b-4b46-815d-03d42f5ee19b">|<img width="327" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/107279507/02715192-4911-44ef-99e0-b90b64d78c09">|
| Desktop |<img width="1007" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/107279507/b13a9611-f6de-4942-97d7-0f0cd979115b">|<img width="1234" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/107279507/15dd0656-16d9-439d-bc55-bf8716a7650d">|

## What areas of the site does it impact?

Mhv medications

## Acceptance criteria

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

Commit:
[MHV-55030] added refill changes behind refill flag.

Commit:
[MHV-55030] unit test.

Commit:
[MHV-55030] added refill changes to prescriptions container behind refill flag.

Commit:
Merge branch 'main' into mhv-55030-update-list-cards

Commit:
[MHV-55030] update unit test.

Commit:
[MHV-55030] removed button type, as the attribute is on the shadow dom button already.

vmenshutin-bylight commented on src/applications/mhv/medications/components/MedicationsList/MedicationsListSort.jsx:
```diff
@@ -38,24 +41,45 @@ const MedicationsListSort = props => {
+              }
+            }}
+          />
+        )}
```
those two buttons look identical to me, pretty sure this could be refactored :)

robertbylight replied:
thank you! I'll go ahead and do that!

Commit:
[MHV-55030] refactored va-button logic.

Commit:
e2e test to verify refill page link on list page

Updated refill submit tests and added new test to verify navigation to refill page from list page

Commit:
Merge branch 'main' into mhv-55030-update-list-cards

Commit:
Merge branch 'main' into mhv-55030-update-list-cards
